File: Common/Objective-C/AgentProxy.m
#import "AgentProxy.h"

@implementation AgentProxy

+ (BOOL)supportsSecureCoding {
    return YES;
}

- (void)encodeWithCoder:(NSCoder *)coder {
    [coder encodeObject:self.text forKey:@"answer"];
    [coder encodeInteger:self.characterCount forKey:@"characterCount"];
    [coder encodeObject:self.timestamp forKey:@"timestamp"];
}

- (id)initWithCoder:(NSCoder *)aDecoder {
	self = [super init];
	if (self) {
        _text = [aDecoder decodeObjectOfClass:[NSString class]
                                         forKey:@"answer"];
        _characterCount = [aDecoder decodeIntegerForKey:@"characterCount"];
        _timestamp = [aDecoder decodeObjectOfClass:[NSDate class]
                                            forKey:@"timestamp"];
	}
	return self;
}

@end

File: Common/Swift/AgentProxy.swift
//  Converted to Swift 5 by Swiftify v5.0.24084 - https://objectivec2swift.com/
import Foundation

public class AgentProxy: NSObject, NSSecureCoding {
    var text = ""
    var characterCount: Int = 0
    var timestamp: Date?

    class var supportsSecureCoding: Bool {
        return true
    }

    func encode(with coder: NSCoder) {
        coder.encode(text, forKey: "answer")
        coder.encode(characterCount, forKey: "characterCount")
        coder.encode(timestamp, forKey: "timestamp")
    }

    required init?(coder aDecoder: NSCoder) {
        super.init()
        text = aDecoder.decodeObjectOfClass(String.self, forKey: "answer") as? String ?? ""
        characterCount = aDecoder.decodeInteger(forKey: "characterCount")
        timestamp = aDecoder.decodeObjectOfClass(Date.self, forKey: "timestamp") as? Date
    }
}

